src/App.js

```diff
@@ -2,77 +2,50 @@ import React, { Component } from 'react';
 import './App.css';
 
 class App extends Component {
-  _isMounted = false;
-
   state = {
-    source: 'https://i.imgur.com/VYUdUNJ.png',
+    source: '',
     error: null,
+    loading: true,
   };
 
-  fetchCat() {
-    fetch("https://aws.random.cat/meow").then(res=>res.json()).then(
-      (result) => {
-        this.setState({
-          source: result.file
-        });
-    },
-    (error) => {
-      this.setState({
-        error
-      });
-    }
-    )
-  }
-
   componentDidMount() {
-    this._isMounted = true;
-    const { source } = this.state;
-    this.fetchCat(source);
-
+    this.fetchCat();
   }
 
-  componentWillUnmount() {
-    this._isMounted = false;
```

Comment:
Removed ComponentWillUnmount, since it wasnt doing anything other than setting a variable